src/azure-cli-core/azure/cli/core/commands/validators.py

```diff
@@ -52,6 +52,7 @@ def get_default_location_from_resource_group(namespace):
 
 def validate_file_or_dict(string):
     import os
+    string = os.path.expanduser(string)
```

Comment:
Rather rollback the change, I prefer  to just use `validate_file_or_dict` for loading extension data, hence i update it to respect the  `~`